swarm/storage/resource.go

```diff
@@ -27,6 +27,30 @@ const (
 	hasherCount         = 8
 )
 
+type ResourceError struct {
+	code int
+	err  string
+}
+
+func (e *ResourceError) Error() string {
+	return e.err
+}
+
+func (e *ResourceError) Code() int {
+	return e.code
+}
+
+func NewResourceError(code int, s string) error {
```

Comment:
this should return `*ResourceError` and look like:

```
if code < 0 || code >= errCnt {
  panic("no such error code")
}
return &ResourceError{
                code: code,
+		err: s,
+	}
```

Reply:
It's implemented on line 44 :)